Upgrade wagtail to 3.x

theskumar opened the pull request:
Fixes #2880
Fixes #2874
Closes #2881 

- Update to wagtail 3.
- Rename BASE_URL to WAGTAILADMIN_BASE_URL
- use_json_field argument added to StreamField
- Change edit handlers to panels and also path to templates
- Make the test_activation_email_includes_link pass
- Use wagtail 3.0.1 and use robust testing for the activation email
- [Add migrations for StreamField, use_json=True](https://github.com/HyphaApp/hypha/pull/2934/commits/273c47eacf44be1c519705bf6cd644a12b0c1208)]
- [Update to wagtail-purge==0.2.0](https://github.com/HyphaApp/hypha/pull/2934/commits/3e8e66cafd4e127299ba49711346e55a321dfdd9)
- [Fix “FilteredFieldPanel” for wagtail 3.x](https://github.com/HyphaApp/hypha/pull/2934/commits/2b76b1b9041de38b4d827ba36b8321d6a9b5e563)
- [Fix “ReadOnlyPanel” and “ReadOnlyInlinePlanel” for wagtail 3.x](https://github.com/HyphaApp/hypha/pull/2934/commits/a9a497f98a70177f20b6bd9c852f313d3ef6b865)

TODOs:
- [x] Fix the test randomly failing
- [x] Review and fix wagtail admin HTML overrides

frjo committed:
Update to wagtail 3.

sks444 committed:
Rename BASE_URL to WAGTAILADMIN_BASE_URL

sks444 committed:
use_json_field argument added to StreamField

sks444 committed:
Change edit handlers to panels and also path to templates

theskumar committed:
Make the test_activation_email_includes_link pass

theskumar committed:
Use wagtail 3.0.1 and use robust testing for the activation email

theskumar committed:
Apply isort

theskumar committed:
Make WAGTAILADMIN_BASE_URL top-level config to be able to deprecate PRIMARY_HOST

theskumar committed:
Run wagtail updatemodulepaths for wagtail 3.x

theskumar committed:
Add migrations for StreamField, use_json=True

theskumar committed:
Run isort

theskumar committed:
Update to wagtail-purge==0.2.0

Changelog: https://github.com/torchbox/wagtail-purge/compare/0.1...main

theskumar committed:
Update to wagtail-cache==2.1.0 

Chagelog: https://github.com/coderedcorp/wagtail-cache/compare/v1.0.2...v2.1.0

theskumar committed:
Update to wagtail-factories==2.1.0

Changelog: https://github.com/wagtail/wagtail-factories/compare/2.0.1...2.1.0

2.1.0
=====
 - Add support for Wagtail 3.0 and drop support for all Wagtail versions
   before 2.15
 - Add support for Django 4.0
 - Add support for Python 3.10
 - Removed support for Python 3.6
 - Removed support for Django 2.2 and 3.1
 - Removed support for factory boy <3.2

theskumar committed:
Fix “FilteredFieldPanel” for wagtail 3.x

theskumar committed:
Fix “ReadOnlyPanel” and “ReadOnlyInlinePlanel” for wagtail 3.x

theskumar committed:
Use robust test for add_screening_status

Do not depend on sorting IDs for checks

theskumar committed:
Fix tests related to Django messages & ReviewForm

theskumar committed:
Move mailchimp/tests.py -> mailchimp/tests/test_views.py

sandeepsajan0 commented on hypha/apply/api/v1/screening/tests/test_views.py:
```diff
@@ -129,13 +129,15 @@ def test_add_screening_status(self):
-        first_submission_screening_status = self.submission.screening_statuses.order_by('-id').last()
+        self.assertEqual(len(response.json()), 2)
+
+        first_submission_screening_status = self.yes_default_screening_status
```
I think we need to revert this line. I understand what you are thinking here but we have a unique logic in its [views here](https://github.com/HyphaApp/hypha/blob/main/hypha/apply/api/v1/screening/views.py#L76) that is responsible to change the screening statuses ids of a submission.

sandeepsajan0 commented on hypha/apply/api/v1/screening/tests/test_views.py:
```diff
@@ -129,13 +129,15 @@ def test_add_screening_status(self):
         self.assertEqual(response.json()[0]['default'], first_submission_screening_status.default)
-        self.assertEqual(response.json()[1]['id'], self.submission.screening_statuses.order_by('id').last().id)
+
+        self.assertEqual(response.json()[1]['id'], self.yes_screening_status.id)
```
We need to change this as well, as per the above comment.

theskumar committed:
Update “StreamFieldPanel” and “ImageChoosePanel” with “FieldPanel”

theskumar committed:
Fix test case for “add_screening_status”

theskumar committed:
chore(apply): fix test case for review form

Use `django.test.TestCase` instead of `unittest.TestCase` to 
enable DB access

theskumar committed:
Relax the testing for Mailchimp library

theskumar committed:
Replace “self.assertEquals” with “self.assertEqual”

theskumar committed:
Fix html for /admin/users/:id/ admin page

theskumar committed:
Fix css styling for input type radio/checkbox

theskumar committed:
Fix warning for django-filter -> “filter_class”


See: https://django-filter.readthedocs.io/en/stable/guide/migration.html#view-attributes-renamed-867

```
hypha/apply/api/v1/screening/tests/test_views.py::ScreeningStatusViewSetTests::test_staff_can_list_screening_statuses
hypha/apply/api/v1/screening/tests/test_views.py::ScreeningStatusViewSetTests::test_staff_can_view_screening_statuses_detail
  /Users/theskumar/work/hyphaapp/hypha/venv/lib/python3.9/site-packages/django_filters/rest_framework/backends.py:31: MigrationNotice: `ScreeningStatusViewSet.filter_class` attribute should be renamed `filterset_class`. See: https://django-filter.readthedocs.io/en/master/guide/migration.html See: https://django-filter.readthedocs.io/en/master/guide/migration.html
    filterset_class = self.get_filterset_class(view, queryset)
```

theskumar committed:
RemovedInDjango40Warning - The {% ifequal %} template tag is deprecated in favor of {% if %}



```
hypha/apply/determinations/tests/test_views.py: 20 warnings
  /Users/theskumar/work/hyphaapp/hypha/venv/lib/python3.9/site-packages/django/template/defaulttags.py:862: RemovedInDjango40Warning: The {% ifequal %} template tag is deprecated in favor of {% if %}.
    warnings.warn(
```

theskumar commented:
The test_add_screening_status is failing for me locally. The reason seems to be there is no pre-defined ordering on the tables, but you are expecting an `ordering` in the returned response objects.

![image](https://user-images.githubusercontent.com/236356/181483486-5f93cfd0-4939-43fa-9982-9bfa9a1be134.png)

I’m not sure how it’s passing in the circle CI, but it’s failing on my machine. @frjo @sandeepsajan0

theskumar committed:
Refractor messaging to separate out it’s adaptors

theskumar committed:
Update flake8 to 5.0.1

frjo reviewed (approved):
Excellent work! Deploying to test now.

theskumar committed:
Merge pull request #2938 from HyphaApp/refractor/messaging

Related to #2882

The messaging.py has quite a lot of code with multiple concerns, which can be easily divided into individual independent modules.

This refractor puts different messaging adapters into its modules for better maintainability and readability.

theskumar committed:
Fix test case for screening API 

The test checks for the ordering of the item returned.
The view function or the model class doesn’t have any
order specified.

This cause the tests to fail randomly on different machines.
This commit removes that ordering the check and just checks for
the availability of the ids.
